### PR TITLE
Merge RewardDistribution with RewardDistributionEntry

### PR DESCRIPTION
**Pull request by ajsantander**

### CODE REVIEW

File: protocol/synthetix/storage.dump.sol [hunk 1]
```diff
@@ -495,24 +495,18 @@ library Pool {
 library RewardDistribution {
     struct Data {
         address distributor;
-        RewardDistributionEntry.Data entry;
+        uint128 __slotAvailableForFutureUse;
```
Comment (noahlitvin) [inline protocol/synthetix/storage.dump.sol:498]: uint128 is the same size?

File: protocol/synthetix/storage.dump.sol [hunk 2]
```diff
@@ -495,24 +495,18 @@ library Pool {
...
         uint128 rewardPerShareD18;
-        mapping(uint256 => RewardDistributionStatus.Data) actorInfo;
-    }
-}
-
-// @custom:artifact contracts/storage/RewardDistributionEntry.sol:RewardDistributionEntry
-library RewardDistributionEntry {
-    struct Data {
+        mapping(uint256 => RewardDistributionClaimStatus.Data) actorInfo;
```
Comment (noahlitvin) [inline protocol/synthetix/storage.dump.sol:500]: Rename from actorInfo here to claimStatus?

File: protocol/synthetix/contracts/storage/RewardDistribution.sol [hunk 3]
```diff
@@ -1,17 +1,170 @@
...
+        uint128 __slotAvailableForFutureUse;
+        /**
+         * @dev The value of the rewards in this entry.
+         */
         uint128 rewardPerShareD18;
-        mapping(uint256 => RewardDistributionStatus.Data) actorInfo;
+        /**
+         * @dev The status for each actor, regarding this distribution's entry.
+         */
+        mapping(uint256 => RewardDistributionClaimStatus.Data) actorInfo;
```
Comment (noahlitvin) [inline protocol/synthetix/contracts/storage/RewardDistribution.sol:39]: Rename to claimStatus?

File: protocol/synthetix/contracts/storage/RewardDistribution.sol [hunk 4]
```diff
@@ -1,17 +1,170 @@
...
-        // 3rd party smart contract which holds/mints the pools
+        /**
+         * @dev The 3rd party smart contract which holds/mints tokens for distributing rewards to vault participants.
+         */
         IRewardDistributor distributor;
-        RewardDistributionEntry.Data entry;
+        /**
+         * @dev Available slot.
+         */
+        uint128 __slotAvailableForFutureUse;
```
Comment (noahlitvin) [inline protocol/synthetix/contracts/storage/RewardDistribution.sol:31]: Is uint128 the right size?

File: protocol/synthetix/contracts/storage/CollateralConfiguration.sol [hunk 5]
```diff
@@ -119,6 +169,11 @@ library CollateralConfiguration {
         }
     }
 
+    /**
+     * @dev Returns the price of this collateral configuration object.
+     * @param self The CollateralConfiguration object.
+     * @return The price of the collateral.
```
Comment (noahlitvin) [inline protocol/synthetix/contracts/storage/CollateralConfiguration.sol:175]: Specify this is 18 decimal precision?

File: protocol/synthetix/contracts/storage/CollateralConfiguration.sol [hunk 6]
```diff
@@ -38,33 +46,61 @@ library CollateralConfiguration {
...
-        /// address which should be used for transferring this collateral
+        /**
+         * @dev The token address for this collateral type.
+         */
         address tokenAddress;
-        /// minimum delegation amount (other than 0), to prevent sybil/attacks on the system due to new entries.
+        /**
+         * @dev Minimum amount that accounts can delegate to pools.
+         * Helps prevent spamming on the system.
+         * Note: If zero, no minimum amount will be used.
```
Comment (noahlitvin) [inline protocol/synthetix/contracts/storage/CollateralConfiguration.sol:83]: I thought we use the liquidation reward amount if this is less than it?

File: protocol/synthetix/contracts/storage/CollateralConfiguration.sol [hunk 7]
```diff
@@ -38,33 +46,61 @@ library CollateralConfiguration {
...
-        /// accounts cannot mint sUSD if their debt brings their cratio below this value
+        /**
+         * @dev System-wide collateralization ratio for issuance of snxUSD.
+         * Accounts will not be able to mint snxUSD if they are below this issuance c-ratio.
+         */
         uint256 issuanceRatioD18;
-        /// accounts below the ratio specified here are immediately liquidated
+        /**
+         * @dev System-wide collateralization ratio for liquidations.
+         * Accounts below this c-ratio can will be immediately liquidated.
```
Comment (noahlitvin) [inline protocol/synthetix/contracts/storage/CollateralConfiguration.sol:65]:
```suggestion
         * Accounts below this c-ratio can be immediately liquidated.
```

File: protocol/synthetix/contracts/storage/CollateralConfiguration.sol [hunk 8]
```diff
@@ -38,33 +46,61 @@ library CollateralConfiguration {
...
         bool depositingEnabled;
-        /// accounts cannot mint sUSD if their debt brings their cratio below this value
+        /**
+         * @dev System-wide collateralization ratio for issuance of snxUSD.
+         * Accounts will not be able to mint snxUSD if they are below this issuance c-ratio.
+         */
         uint256 issuanceRatioD18;
-        /// accounts below the ratio specified here are immediately liquidated
+        /**
+         * @dev System-wide collateralization ratio for liquidations.
```
Comment (noahlitvin) [inline protocol/synthetix/contracts/storage/CollateralConfiguration.sol:64]:
```suggestion
         * @dev System-wide collateralization ratio for liquidations of this collateral type.
```

File: protocol/synthetix/storage.dump.sol (hunk 1 above)
Comment (ajsantander) [reply protocol/synthetix/storage.dump.sol:498]: No, the removed var is a bytes32 pointer, so I am effectively doing a non backwards compatible change here.

The var I keep there is to make use of the half word that's free from the next uint128.

File: protocol/synthetix/contracts/storage/CollateralConfiguration.sol (hunk 6 above)
Comment (ajsantander) [reply protocol/synthetix/contracts/storage/CollateralConfiguration.sol:83]: And you are right! Fixed 🙏

File: protocol/synthetix/contracts/storage/CollateralConfiguration.sol (hunk 7 above)
Comment (ajsantander) [reply protocol/synthetix/contracts/storage/CollateralConfiguration.sol:65]: Will include in my commit 👍

File: protocol/synthetix/contracts/storage/CollateralConfiguration.sol (hunk 8 above)
Comment (ajsantander) [reply protocol/synthetix/contracts/storage/CollateralConfiguration.sol:64]: Will include in my commit 👍

**Review by noahlitvin — APPROVED** [review]

**Pull request merged**